src/components/seeds/database/Filters.tsx

```diff
@@ -56,33 +57,49 @@ const useStyles = makeStyles((theme: Theme) => ({
   },
   filtersContainer: {
     minHeight: '32px',
-    flex: (props: StyleProps) => (props.isMobile ? '1 0' : '8 0 600px'),
+    display: 'flex',
+    flexDirection: (props: StyleProps) => (props.isMobile ? 'column' : 'row'),
+    flexWrap: (props: StyleProps) => (props.isMobile ? 'nowrap' : 'wrap'),
+    alignItems: (props: StyleProps) => (props.isMobile ? 'flex-start' : 'center'),
+    gap: theme.spacing(1.5),
     marginTop: `-${theme.spacing(1)}`,
   },
+  searchField: {
+    width: '300px',
+    marginTop: theme.spacing(-0.5),
+  },
 }));
 
 interface Props {
   columns: DatabaseColumn[];
+  searchColumns: DatabaseColumn[];
   filters: Record<string, SearchNodePayload>;
   availableValues: FieldValuesPayload;
   allValues: FieldValuesPayload;
   onChange: (filters: Record<string, SearchNodePayload>) => void;
 }
 
 export default function Filters(props: Props): JSX.Element {
+  const { columns, searchColumns, filters, availableValues, allValues, onChange } = props;
   const { isMobile, isDesktop } = useDeviceInfo();
   const classes = useStyles({ isMobile, isDesktop });
   const [popover, setPopover] = React.useState<FilterPopover>();
+  const [searchTerm, setSearchTerm] = React.useState('');
+  const debouncedSearchTerm = useDebounce(searchTerm, 250, (value) => {
+    const updatedSearchFilters = getSearchTermFilter(searchColumns, value);
+    onChange({ ...filters, ...updatedSearchFilters });
```

Comment:
callback is fine.. thoughts on useEffect on the debounced search term, just to keep with the react-ive pattern (vs javascript callback patterns). tbh i don't know if there are pros/cons.. more of an open comment really.
also just curious if the value of this callback changes on each render (since it's a dependency inside useDebounce's useEffect).. if so, we could make it a useCallback or something like that.. mostly curious here...

Reply:
I was originally trying to use a `useEffect` with basically this callback as the effect. When I did that, though, I ended up needing `filters` as a dependency of the effect. This caused an infinite recursion as calling `onChange` updated filters, which called effect and so on. This callback was the most straightforward way I could think of out of that situation.

Reply:
>curious if the value of this callback changes on each render

I don't think so! So having a useCallback might make sense!

Reply:
Got it, no worries.. I guess the callback approach makes sense then.